setendings.py

```diff
@@ -0,0 +1,397 @@
+                        node.as_string())
+    else:
+        node.end_lineno = node.tolineno
+
+def _set_end_col_offset_by_string(node):
+    """Nodes without children can get end_col_offset by length of string.
+    Hopefully this works without problems.
+    """
+    NO_CHILDREN_TYPE = [astroid.Arguments, astroid.AssignName, astroid.Break, 
+            astroid.Const, astroid.DelName, astroid.Ellipsis, astroid.Global, 
+            astroid.Import, astroid.ImportFrom, astroid.List, astroid.Name, 
+            astroid.Nonlocal, astroid.Pass, astroid.Yield]
+
+    node_string = node.as_string()
+
+    if type(node) not in NO_CHILDREN_TYPE:
+        raise Exception('''ERROR:️ node not found in NO_CHILDREN list.''', node, 
+                        node_string)
+
+    # Guard: to use this approach, it shouldn't have children.
+    for child in node.get_children():  # get_children() returns a generator.
+        raise Exception('''ERROR:️ should not be called if has children.''', 
+                        node, node_string)
+
+    # Guard: should have .value property
+    if not hasattr(node, 'as_string'):
+        raise Exception('''ERROR:️ node must have the .as_string property.''', 
+                        node, node_string)
+
+    # _set_child_end_col_offset_by_value(node)
+    node.end_col_offset = node.col_offset + len(node_string)
+
+def _set_end_col_offset(node):
+    """Set the end_col_offset property by the as_string attribute.
+    Uses the end column location of the last line, instead of the widest line.
+    Need to do a custom postorder traversal to set end_col_offset property of 
+    child nodes first, since it is sometimes unset in the last child.
+    Postorder setting of children results in the assumption: if a node has the
+    propert set, then all of its children are set.
+    """
+
+    # Compare with "None" since value could be "0", which is falsey.
+    # Returning a.s.a.p. reduces the runtime.
+    # Some nodes dont have the attribute set yet.
+    if hasattr(node, 'end_lineno') and hasattr(node, 'end_col_offset'):
+        if node.end_lineno is not None and node.end_col_offset is not None:
+            return  # since properties are already set.
+
+    # Doing this first makes it a postorder traversal.
+    for child in node.get_children():  # get_children() returns a generator.
+        _set_end_col_offset(child)
+
+    # if possible, set the end_col_offset by that of the last child
+    if node.last_child():
+        _set_by_last_child(node, node.last_child())
+    # Arrgghh astroid.Arguments cannot rely on last_child() being set properly.
+    elif type(node) is astroid.Arguments:
+        last_child = None
+        for last_child in node.get_children():  # get_children() returns a generator
+            pass  # skip to last
+        print('last item in Arguments:', last_child)
+        _set_by_last_child(node, last_child)
+    # No children. Postorder traversal sets these children first.
+    else:
```

Comment:
This, too, can be split into a separate helper function.

In general, you can `register_transform` this function to the nodes which are guaranteed to have a `last_child` method (right now, just `BinOp`), and `register_transform` the `_set_end_col_offset_by_string` function to the ones that don't (right now, just `Const`).